### PR TITLE
add motan plugin support

### PR DESCRIPTION
**Pull request by zeying-wang**
- support to the latest version

### CODE REVIEW

**Comment by observeralone** [comment]
please base master to code and pr
you can use following command to rebase:
```bash
git checkout master
git pull origin master:master
git checkout {you branch}
git rebase -i master
```

**Review by landyking — APPROVED** [review]
LGTM

File: plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/provider/MotanProviderTraceInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Request request = (Request) methodInfo.getArgs()[0];
+        Provider<?> provider = (Provider<?>) methodInfo.getArgs()[1];
+        MotanCtxUtils.initProviderSpan(context, provider.getUrl(), request);
+    }
+
+    @Override
+    public void after(MethodInfo methodInfo, Context context) {
+        Response response = (Response) methodInfo.getRetValue();
+        Throwable throwable = methodInfo.getThrowable();
+        MotanCtxUtils.finishProviderSpan(response, throwable, context);
```
Comment (observeralone) [inline plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/provider/MotanProviderTraceInterceptor.java:34]: Please make sure the `Scope` has been closed in the `after` method.

File: plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/consumer/MotanConsumerTraceInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	public void after(MethodInfo methodInfo, Context context) {
+		Response response = (Response) methodInfo.getRetValue();
+		if (MotanClassUtils.DefaultResponseFutureTypeChecker.getTypeChecker().hasClassAndIsType(response)) {
+			DefaultResponseFuture defaultResponseFuture = (DefaultResponseFuture) response;
+			defaultResponseFuture.addListener(new TraceFutureListener(context.exportAsync()));
+		} else {
+			MotanCtxUtils.finishConsumerSpan(response, methodInfo.getThrowable(), context);
+		}
```
Comment (observeralone) [inline plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/consumer/MotanConsumerTraceInterceptor.java:37]: Please make sure the `Scope` has been closed in the `after` method.

File: plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/consumer/MotanConsumerTraceInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		Request request = (Request) methodInfo.getArgs()[0];
+		MotanCtxUtils.initConsumerSpan(context, url, request);
+	}
+
+	@Override
+	public void after(MethodInfo methodInfo, Context context) {
+		Response response = (Response) methodInfo.getRetValue();
+		if (MotanClassUtils.DefaultResponseFutureTypeChecker.getTypeChecker().hasClassAndIsType(response)) {
+			DefaultResponseFuture defaultResponseFuture = (DefaultResponseFuture) response;
+			defaultResponseFuture.addListener(new TraceFutureListener(context.exportAsync()));
```
Comment (observeralone) [inline plugins/motan/src/main/java/com/megaease/easeagent/plugin/motan/interceptor/trace/consumer/MotanConsumerTraceInterceptor.java:34]: `Scope` must close whether it is asynchronous or not

File: README.md [hunk 4]
```diff
@@ -58,6 +58,8 @@ A lightweight & opening Java Agent for Cloud-Native and APM system
         * `Jedis 3.5.x`、 `Lettuce 5.3.x (sync、async)`
         * `ElasticSearch Client >= 7.x (sync、async)`
         * `Mongodb Client >=4.0.x (sync、async)`
+        * `Dubbo >=2.5.x`
```
Comment (observeralone) [inline README.md:61]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -58,6 +58,7 @@ A lightweight & opening Java Agent for Cloud-Native and APM system
         * `Jedis 3.5.x`、 `Lettuce 5.3.x (sync、async)`
         * `ElasticSearch Client >= 7.x (sync、async)`
         * `Mongodb Client >=4.0.x (sync、async)`
+        * `Motan`
```
Comment (observeralone) [inline README.md:61]:
```suggestion
        * `Motan >= 1.2.0`
```

File: plugins/motan/pom.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>motan</artifactId>
+
+    <properties>
+        <maven.compiler.source>18</maven.compiler.source>
+        <maven.compiler.target>18</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <motan.version>LATEST</motan.version>
```
Comment (observeralone) [inline plugins/motan/pom.xml:18]:
```suggestion
        <motan.version>1.2.0</motan.version>
```

**Review by observeralone — APPROVED** [review]

**Pull request merged**